WW3_PreProc/ww3_grid_inp2nml.sh

```diff
@@ -0,0 +1,2181 @@
+declare -A slope2d
+declare -A sed2d
+
+declare -A inptx
+declare -A inpty
+declare -A inptf
+declare -A exptx
+declare -A expty
+declare -A exptf
+declare -A exbdx
+declare -A exbdy
+declare -A oulnx0
+declare -A oulny0
+declare -A oulndx
+declare -A oulndy
+declare -A oulnnp
+
+#------------------------------
+# clean up inp file from all $ lines
+
+cleaninp="$cur_dir/${prog}_clean.inp"
+rm -f $cleaninp
+
+cat $inp | while read line
+do
+
+  if [ "$(echo $line | cut -c1)" = "$" ]
+  then
+    continue
+  fi
+
+  cleanline="$(echo $line | awk -F' ' '{print $1}' | sed -e "s/\*//g" -e "s/\"//g" -e "s/\'//g")"  
+  if [ -z "$cleanline" ]
+  then
+    continue
+  fi
+
+  echo "$line" >> $cleaninp
+
+done
+
+
+
+#------------------------------
+# get all values from clean inp file
+
+readarray -t lines < "$cleaninp"
+numlines=${#lines[@]}
+
+# remove carriage return characters
+for il in $(seq 0 $numlines)
+do
+  lines[$il]=$(echo "$(echo ${lines[$il]} | sed 's/\r$//')")
+done
+il=0
+
+# grid name
+gridnameinp=$(echo $(echo "${lines[$il]}" | sed -e "s/\*//g" -e "s/\"//g" -e "s/'//g"))
+echo $gridnameinp
+
+if [ "$(basename $inp)" == "ww3_grid.inp" ]
+then
+  grdname="$(echo $gridnameinp | awk -F' ' '{print $1}')"
+  echo 'grdname from grid name : ' "$grdname"
+else
+  baseinp=$(basename $inp)
+  grdname="$(echo ${baseinp%%.*}  | awk -F'ww3_grid_' '{print $2}')"
+  echo 'grdname from inp name : ' "$grdname"
+fi
+
+# spectrum
+il=$(($il+1))
+xfr="$(echo ${lines[$il]} | awk -F' ' '{print $1}' | cut -d \" -f2  | cut -d \' -f2)"
+freq1="$(echo ${lines[$il]} | awk -F' ' '{print $2}' | cut -d \" -f2  | cut -d \' -f2)"
+nk="$(echo ${lines[$il]} | awk -F' ' '{print $3}' | cut -d \" -f2  | cut -d \' -f2)"
+nth="$(echo ${lines[$il]} | awk -F' ' '{print $4}' | cut -d \" -f2  | cut -d \' -f2)"
+thoff="$(echo ${lines[$il]} | awk -F' ' '{print $5}' | cut -d \" -f2  | cut -d \' -f2)"
+echo 'spec : ' $xfr $freq1 $nk $nth $thoff
+
+# run
+il=$(($il+1))
+fldry="$(echo ${lines[$il]} | awk -F' ' '{print $1}' | cut -d \" -f2  | cut -d \' -f2)"
+flcx="$(echo ${lines[$il]} | awk -F' ' '{print $2}' | cut -d \" -f2  | cut -d \' -f2)"
+flcy="$(echo ${lines[$il]} | awk -F' ' '{print $3}' | cut -d \" -f2  | cut -d \' -f2)"
+flcth="$(echo ${lines[$il]} | awk -F' ' '{print $4}' | cut -d \" -f2  | cut -d \' -f2)"
+flck="$(echo ${lines[$il]} | awk -F' ' '{print $5}' | cut -d \" -f2  | cut -d \' -f2)"
+flsou="$(echo ${lines[$il]} | awk -F' ' '{print $6}' | cut -d \" -f2  | cut -d \' -f2)"
+echo 'flags : ' $fldry $flcx $flcy $flcth $flck $flsou
+
+# timesteps
+il=$(($il+1))
+dtmax="$(echo ${lines[$il]} | awk -F' ' '{print $1}' | cut -d \" -f2  | cut -d \' -f2)"
+dtxy="$(echo ${lines[$il]} | awk -F' ' '{print $2}' | cut -d \" -f2  | cut -d \' -f2)"
+dtkth="$(echo ${lines[$il]} | awk -F' ' '{print $3}' | cut -d \" -f2  | cut -d \' -f2)"
+dtmin="$(echo ${lines[$il]} | awk -F' ' '{print $4}' | cut -d \" -f2  | cut -d \' -f2)"
+echo 'timesteps : ' $dtmax $dtxy $dtkth $dtmin
+
+# namelists
```

Comment:
looks like you're modifying a nml here - would https://github.com/marshallward/f90nml be useful for that?

Reply:
`ww3_grid_inp2nml.sh` script is downloaded from the WW3 repository to convert [`ww3_grid.inp`](https://github.com/NOAA-EMC/WW3/blob/develop/model/inp/ww3_grid.inp) to [`ww3_grid.nml`](https://github.com/NOAA-EMC/WW3/blob/develop/model/nml/ww3_grid.nml).  I propose directly using `ww3_grid.nml` in the configuration instead of the two-step process with `ww3_grid.inp`.

Reply:
I have added `ww3_grid.nml` and `namelists_Global.nml` files to the `WW3_PreProc` directory (Refer this commit f8ca8cf4fa9858125ab0ab5d9edcd12eddd09399). 

`ww3_grid.inp` and `ww3_grid_inp2nml.sh` is not required now, we can remove it.